### ISSUE TITLE
训练阶段1的损失函数

### ISSUE THREAD

**Issue by Afreshbird** [Zhaozixiang1228/MMIF-CDDFuse#32]
您好，感谢您杰出的工作，我有一个损失函数方面的问题想要请教：

我在代码中看到阶段1的损失函数定义如下：
loss = coeff_mse_loss_VF * mse_loss_V + coeff_mse_loss_IF * mse_loss_I  + coeff_decomp * loss_decomp + coeff_tv * Gradient_loss
与论文中描述的损失函数相比，这里是否多了一项 coeff_tv * Gradient_loss ？

并且我看到 Gradient_loss 的定义如下：
Gradient_loss = L1Loss(kornia.filters.SpatialGradient()(data_VIS), kornia.filters.SpatialGradient()(data_VIS_hat))
如果这一项不是多余的，那么为什么只计算了可见光图像的此项损失而没有计算红外图像的此项损失呢？

**Comment by zxp771** [comment]
感谢赵博士分享您的工作，与这位一样，我也对loss函数有疑问。按照您论文中的叙述，Gradient_loss应该是被用在二阶段的训练而非一阶段的训练，但代码的实现确实相反的。想问一下具体的原因是什么，以及为何现在放在一阶段训练时使用的Gradient_loss只对可见光图像进行了计算。

**Comment by zxp771** [comment]
@Afreshbird 您好，我也有一样的疑问。您有尝试去掉这项损失么？训练效果如何？

**Comment by Afreshbird** [comment]
> @Afreshbird 您好，我也有一样的疑问。您有尝试去掉这项损失么？训练效果如何？

我没去，没去复现的精度和论文基本一致。

**Comment by zxp771** [comment]
@Afreshbird 感谢您的回复

**Comment by eecoder-dyf** [comment]
> > @Afreshbird 您好，我也有一样的疑问。您有尝试去掉这项损失么？训练效果如何？
> 
> 我没去，没去复现的精度和论文基本一致。

你好，请问你复现时有发现stage I中的DIDF_Decoder多了一项data_VIS/data_IR的输入么，貌似与论文不符

**Comment by Pursuefreedomyuan** [comment]
> > @Afreshbird 您好，我也有一样的疑问。您有尝试去掉这项损失么？训练效果如何？
> 
> 我没去，没去复现的精度和论文基本一致。

@Afreshbird 您好，在复现这篇论文的时候batch_size设置的是多少呢，论文中说的是16，但是代码中的是8

**Comment by Afreshbird** [comment]
> > > @Afreshbird 您好，我也有一样的疑问。您有尝试去掉这项损失么？训练效果如何？
> > 
> > 
> > 我没去，没去复现的精度和论文基本一致。
> 
> @Afreshbird 您好，在复现这篇论文的时候batch_size设置的是多少呢，论文中说的是16，但是代码中的是8

用的代码里的（我个人觉得batch_size的影响不大）。
